File: 3838-weighted-word-mapping/3838-weighted-word-mapping.cpp
class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string ans="";
        for(string &word:words){
            int wt=0;
            for(char &ch:word){
                wt+=weights[ch-'a'];
            }
            wt%=26;
            ans+='z'-wt;
        }
        return ans;
    }
};

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder ans=new StringBuilder();
        for(String word:words){
            int wt=0;
            for(char ch:word.toCharArray()){
                wt+=weights[ch-'a'];
            }
            wt%=26;
            ans.append((char)('z'-wt));
        }
        return ans.toString();
    }
}